FiniteStateMachine/src/statemachine/State.java

```diff
@@ -2,11 +2,9 @@
 
 import org.jetbrains.annotations.NotNull;
 
-import java.util.EnumSet;
+import java.util.Set;
 
 
-public interface State<E extends Enum<E> & State<E>> {
```

Comment:
part of the reason for the `Enum` type requirement was to force the actual states to be defined entirely at compile time, static, and in one class. Enums aren't extensible, nor are they anything but static.